app/services/declarations/create.rb

```diff
@@ -0,0 +1,166 @@
+    end
+
+    def participant
+      @participant ||= begin
+        Participants::Query.new(lead_provider:).participant(ecf_id: participant_id)
+      rescue ActiveRecord::RecordNotFound
+        nil
+      end
+    end
+
+  private
+
+    attr_writer :raw_declaration_date
+
+    delegate :schedule, to: :application, allow_nil: true
+    delegate :cohort, to: :schedule
+
+    def declaration_parameters
+      {
+        declaration_date:,
+        declaration_type:,
+        lead_provider:,
+      }
+    end
+
+    def existing_declaration
+      @existing_declaration ||= participant
+        .declarations
+        .joins(application: :course)
+        .submitted_state
+        .or(
+          participant
+            .declarations
+            .joins(application: :course)
+            .billable,
+        )
+        .find_by(declaration_parameters.merge(application: { courses: { identifier: course_identifier } }))
+    end
+
+    def find_or_create_declaration!
+      @declaration = existing_declaration || Declaration.create!(declaration_parameters.merge(application:, cohort:))
+    end
+
+    def statement_attacher
+      @statement_attacher ||= StatementAttacher.new(declaration:)
+    end
+
+    def output_fee_statement_available
+      return if errors.any?
+      return if application.blank?
+      return if existing_declaration&.submitted_state?
+      return if existing_declaration.nil? && !application.fundable?
+      return if lead_provider.next_output_fee_statement(cohort).present?
+
+      errors.add(:cohort, :no_output_fee_statement, cohort: cohort.start_year)
+    end
+
+    def set_eligibility!
+      if declaration.duplicate_declarations.any?
+        declaration.update!(superseded_by: original_declaration)
+        declaration.ineligible_state!
+      elsif application.fundable?
+        declaration.eligible_state!
+      end
+    end
+
+    def validate_schedule_exists
+      return if errors.any?
+      return if schedule&.allowed_declaration_types&.include?(declaration_type)
+
+      errors.add(:declaration_type, :mismatch_declaration_type_for_schedule)
+    end
+
+    def original_declaration
+      @original_declaration ||= declaration.duplicate_declarations.order(created_at: :asc).first
+    end
+
+    def validates_billable_slot_available
+      return if errors.any?
+      return unless participant
+
+      return unless Declaration
+                      .joins(application: %i[user course])
+                      .where(
+                        application: { user: participant, courses: { identifier: course_identifier } },
+                        declaration_type:,
+                        state: %w[submitted eligible payable paid],
+                      ).exists?
+
+      errors.add(:base, :declaration_already_exists)
+    end
+
+    def validate_has_passed_field
+      self.has_passed = has_passed.to_s
+
+      return unless has_passed.blank? || !%w[true false].include?(has_passed)
```

Comment:
in ECF we return different messages for blank vs wrong values, should we do the same here?

Reply:
Hi @cwrw Yeah I noticed that.. but the message content is the same that's why I merged both here..

https://github.com/DFE-Digital/early-careers-framework/blob/main/config/locales/en.yml#L60
https://github.com/DFE-Digital/early-careers-framework/blob/main/config/locales/en.yml#L81